code/datums/atmosphere/_atmosphere.dm

```diff
@@ -19,7 +19,10 @@
 /datum/atmosphere/proc/generate_gas_string()
 	var/target_pressure = rand(minimum_pressure, maximum_pressure)
 	var/pressure_scalar = target_pressure / maximum_pressure
-	
+
+	if(HAS_TRAIT(SSstation, STATION_TRAIT_UNNATURAL_ATMOSPHERE))
+		restricted_chance = min(restricted_chance + 40, 100)
```

Comment:
You don't even really need to min this number, but I will say that 40 is a bit high. It'd set the chance to 90, which means mostly restricted gasses. Kinda worried about a hyper plasma saturated lavaland/icemoon

Reply:
I like the idea of a hyper plasma saturated lavaland/icemoon.